contracts/smart-account/SmartAccount.sol

```diff
@@ -127,10 +127,19 @@ contract SmartAccount is
             revert CallerIsNotAnEntryPoint(msg.sender);
         }
 
-        (, address validationModule) = abi.decode(
-            userOp.signature,
-            (bytes, address)
-        );
+        address validationModule;
```

Comment:
`address validationModule = address(bytes20(userOp.signature[44:64]));`

saves about 160 gas more

In theory, when making userOp.signature, we can do `encodePacked` on the validator address and signature rather than `encode`, to avoid extra trailing 0s.
However, it would require changes in SDK. 

How much effort would it take @livingrockrises @AmanRaj1608 to change this?

This would allow us to more predictably slice userOp.signature rather than decode it.

Reply:
Ah, userOp is in calldata so I should've done a calldataload instead of needlessly copying the signature to memory and mloading.
Good catch

Reply:
Decided to introduce this change in v3 for sure, and maybe in later 2.x versions. But not in 2.1 as it would require a lot of deployments